### PR TITLE
Add hvpa for Prometheus deployed by seed-bootstrap

### PR DESCRIPTION
**Pull request by wyb1**
**How to categorize this PR?**

/area monitoring
/kind enhancement

**What this PR does / why we need it**:

Add HVPA for all prometheus instances managed by `seed-bootstrap`

**Which issue(s) this PR fixes**:
Partial fix for #3889

Once hvpa is rolled out for the Prometheus in the shoot control plane the above issue will be fixed.

**Special notes for your reviewer**:

I kept the original vpa objects in case hvpa is disabled.

**Release note**:

```other operator
Add HVPA for all prometheus instances managed by `seed-bootstrap`
```

### CODE REVIEW

File: pkg/operation/seed/seed.go [hunk 1]
```diff
@@ -350,6 +350,18 @@ func RunReconcileSeedFlow(
...
+		resources := []client.Object{
+			&autoscalingv1beta2.VerticalPodAutoscaler{ObjectMeta: metav1.ObjectMeta{Name: "prometheus-vpa", Namespace: v1beta1constants.GardenNamespace}},
+			&autoscalingv1beta2.VerticalPodAutoscaler{ObjectMeta: metav1.ObjectMeta{Name: "aggregate-prometheus-vpa", Namespace: v1beta1constants.GardenNamespace}},
+		}
+
+		for _, resource := range resources {
+			if err := k8sSeedClient.Client().Delete(ctx, resource); client.IgnoreNotFound(err) != nil {
+				return err
+			}
+		}
```
Comment (rfranzke) [inline pkg/operation/seed/seed.go]: You could use `kutil.DeleteObjects(...)`

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: pkg/operation/seed/seed.go [hunk 2]
```diff
@@ -350,6 +350,16 @@ func RunReconcileSeedFlow(
...
 			return err
 		}
+	} else {
+		// Clean up stale vpa objects
+		resources := []client.Object{
+			&autoscalingv1beta2.VerticalPodAutoscaler{ObjectMeta: metav1.ObjectMeta{Name: "prometheus-vpa", Namespace: v1beta1constants.GardenNamespace}},
+			&autoscalingv1beta2.VerticalPodAutoscaler{ObjectMeta: metav1.ObjectMeta{Name: "aggregate-prometheus-vpa", Namespace: v1beta1constants.GardenNamespace}},
+		}
+
+		if err := kutil.DeleteObjects(ctx, k8sSeedClient.Client(), resources...); client.IgnoreNotFound(err) != nil {
```
Comment (ialidzhikov) [inline pkg/operation/seed/seed.go]: There is no need for `client.IgnoreNotFound(err) != nil` as this check is already done in `DeleteObject` func. See https://github.com/gardener/gardener/blob/779eb21a6118f81d1e0cb2fd92463158cdc643f6/pkg/utils/kubernetes/object.go#L47

```suggestion
		if err := kutil.DeleteObjects(ctx, k8sSeedClient.Client(), resources...); err != nil {
```

**Review by ialidzhikov — COMMENTED** [review]
/lgtm
/needs second-opinion

Let's hold this one until v1.21.0 is released
/hold

**Comment by Kristian-ZH** [comment]
I am not sure if this is still relevant, but when we added HVPA for Loki, we should also set dynamically the resources recommended by the VPA into the Sts's chart: 

https://github.com/gardener/gardener/blob/master/pkg/operation/seed/seed.go#L461 // for garden Loki
https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/logging.go#L66 // for shoot Lokis

Without this change, the Gardenlet did not know about the updates and was reverting them.

**Comment by rfranzke** [comment]
Good point @Kristian-ZH, thank you!
/author-action

**Comment by wyb1** [comment]
Thanks for the feedback! PTAL
Note: the templating doesn't like it when there is a `-` in the Values so I used a workaround with `Index`
```
{{- index .Values.prometheus.resources "prometheus-config-reloader" | toYaml | nindent 10 }}
```

**Review by rfranzke — APPROVED** [review]
/lgtm
/invite @Kristian-ZH

**Review by Kristian-ZH — APPROVED** [review]
/lgtm

**Pull request merged**